components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -18102,6 +18289,163 @@ public void addOperationPolicyMapping(Set<URITemplate> uriTemplates) throws APIM
         }
     }
 
+    public APIEndpointInfo getAPIEndpoint(String apiUUID, String endpointUUID) throws APIManagementException {
+        APIEndpointInfo apiEndpoint;
+        String sql = SQLConstants.APIEndpointsSQLConstants.GET_API_ENDPOINT_BY_API_UUID_AND_ENDPOINT_UUID;
+        try (Connection conn = APIMgtDBUtil.getConnection()) {
+            int apiId = getAPIID(apiUUID, conn);
+            PreparedStatement ps = conn.prepareStatement(sql);
+            ps.setInt(1, apiId);
+            ps.setString(2, endpointUUID);
+            try (ResultSet rs = ps.executeQuery()) {
+                if (rs.next()) {
+                    apiEndpoint = new APIEndpointInfo();
+                    apiEndpoint.setEndpointUuid(rs.getString("ENDPOINT_UUID"));
```

Comment:
Convert this list of strings also to constants. "ENDPOINT_UUID" etc.

Reply:
this is also DB fetching right?